### ISSUE TITLE
All waiting tasks will be executed at once

### ISSUE THREAD

**Issue by tothegills** [OctopusDeploy/Issues#1801]
By default, Octopus Server will pop and execute all waiting tasks with each pass of the task queue.  If a lot of tasks are queued (think hundreds) this can be detrimental to performance. In clusters this also results in a single server executing all tasks.

**Comment by tothegills** [comment]
The plan is to check resource usage before taking on another task.

**Comment by tothegills** [comment]
The new plan is to limit the number of tasks that can execute at once on each node.

**Comment by tothegills** [comment]
Release note: Limit the number of tasks that can be run in parallel (defaults to 5)
